### PR TITLE
Accessory Loadout Options

### PR DESCRIPTION
**Pull request by Geevies**
* You can now choose whether your loadout accessory spawns on your uniform, your suit, or gets put into the suit slot as a standalone item, such as with capes and ponchos.
* Fixed only one accessory being visible at a time in the loadout preview.
* Fixed vaurca tunnel cloaks not being visible in the loadout preview.

image
image

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Review by alsoandanswer — APPROVED** [review]

File: code/controllers/subsystems/job.dm [hunk 1]
```diff
@@ -735,15 +741,42 @@ SUBSYSTEM_DEF(jobs)
...
-				Debug("ECD/([H]): Unable to equip [thing]; dumping into overflow.")
+			var/obj/item/CI = G.spawn_item(H, metadata, H)
+			var/equip_slot = G.slot
+			var/handled_accessory = FALSE
+
+			if(isaccessory(CI))
+				var/datum/gear_tweak/accessory_slot/accessory_slot = locate(/datum/gear_tweak/accessory_slot) in G.gear_tweaks
+				if(accessory_slot && metadata["[accessory_slot]"])
+					var/selected_slot = metadata["[accessory_slot]"]
+					if(selected_slot)
```
Comment (FluffyGhoster) [inline code/controllers/subsystems/job.dm:753]: This looks redundant with the switch

File: code/controllers/subsystems/job.dm [hunk 2]
```diff
@@ -735,15 +741,42 @@ SUBSYSTEM_DEF(jobs)
...
+										worn_suit.attach_accessory(H, CI)
+										handled_accessory = TRUE
+							if(GEAR_TWEAK_ACCESSORY_SLOT_SUIT_STANDALONE)
+								equip_slot = slot_wear_suit
+
+			if(!handled_accessory)
+				if (H.equip_to_slot_or_del(CI, equip_slot))
+					to_chat(H, SPAN_NOTICE("Equipping you with [thing]!"))
+					used_slots += equip_slot
+					Debug("ECD/([H]): Equipped [thing] successfully.")
```
Comment (FluffyGhoster) [inline code/controllers/subsystems/job.dm:776]: Needs its own logging function module

File: code/controllers/subsystems/job.dm [hunk 3]
```diff
@@ -735,15 +741,42 @@ SUBSYSTEM_DEF(jobs)
...
+				else
+					. += thing
+					Debug("ECD/([H]): Unable to equip [thing]; dumping into overflow.")
```
Comment (FluffyGhoster) [inline code/controllers/subsystems/job.dm:779]: Needs its own logging function module

File: code/modules/client/preference_setup/loadout/loadout.dm [hunk 4]
```diff
@@ -432,24 +434,34 @@ var/list/gear_datums = list()
...
 	for(var/datum/gear_tweak/gt in gear_tweaks)
 		if(metadata["[gt]"])
 			gt.tweak_gear_data(metadata["[gt]"], gd, H)
 		else
 			gt.tweak_gear_data(gt.get_default(), gd, H)
-	if(ispath(gd.path, /obj/item/organ/external))
-		var/obj/item/organ/external/external_aug = gd.path
+	return list(gd.path, gd.location)
+
+/datum/gear/proc/spawn_item(var/location, var/metadata, var/mob/living/carbon/human/H)
```
Comment (FluffyGhoster) [inline code/modules/client/preference_setup/loadout/loadout.dm:447]: Probably should be marked as SHOULD_NOT_SLEEP(TRUE)

Comment (Geevies) [reply code/modules/client/preference_setup/loadout/loadout.dm:447]: tried adding it and it exploded the linter, looks like changes unrelated to this PR so im not gonna bother

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**